src/sessions/SessionManager.ts

```diff
@@ -104,8 +104,8 @@ export class SessionManager {
     /**
      * Returns the session ID. If no session ID exists, one will be created.
      */
-    public getSession(): Session {
-        if (this.session.sessionId === NIL_UUID) {
+    public getSession(createIfNull = true): Session {
+        if (createIfNull && this.session.sessionId === NIL_UUID) {
```

Comment:
Currently, the page is created before a session is created , ensuring that there are page attributes when the session is created and the session start event is recorded. In the case that `getSession()` is called when initializing the web client and it is a _nil_ session, the page hasn't been created so the session start event metadata is missing the page related attributes. This bug was detected by one of the integ tests that was verifying that a session start event was recorded with the `pageId` attribute in the metadata. 

This approach was to keep the current web client behavior of when we create sessions so that we do not introduce the new scenario of prematurely creating a new session during initialization. I'd prefer to not increase the risk of data loss that comes with (1) and (2), especially if we will stop emitting the session start event and we can clean up this variable later. Or is the concern about introducing `createIfNull` when it could become obsolete later?

Reply:
+1 to Quinn's point, we will always be arbitrarily creating the sample rate (thus fetching credentials arbitrarily)

Reply:
Will refactor to make the sampling decision before creating the session in next revision!

Reply:
Refactored so that sampling decision is always evaluated when creating a session. Only exception is to if the existing session is the nil session that hasn't expired - in this case, we'll use existing sampling decision.